packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -243,13 +252,77 @@ export class ApexDebug extends DebugSession {
   protected threadsRequest(response: DebugProtocol.ThreadsResponse): void {
     const debuggedThreads: Thread[] = [];
     for (let threadId = 0; threadId < this.requestThreads.length; threadId++) {
-      debuggedThreads.push(new Thread(threadId, this.requestThreads[threadId]));
+      debuggedThreads.push(
+        new Thread(threadId, `Request ID: ${this.requestThreads[threadId]}`)
```

Comment:
What did we use to display in IDE 1? Could we mimic that here?
I remember that sometimes it can be a URL to an VF page, etc.

Reply:
Here's what it looks like in IDE1:

![image](https://user-images.githubusercontent.com/7286437/29982317-14f92a58-8f06-11e7-8cc9-2caaf4132323.png)

That URL is stored in ApexDebuggerRequest.Description, so we'd have to fetch it when we handle RequestStarted. I didn't want to do it unless we really need it because calling out to the CLI is slow. If I were to add that, I would explore just using the CLI for access token, then use request-light for everything else.

Reply:
Could you keep a list of things that would be more snappy if we could save a call to the CLI?
I want to surface this to our CLI team and discuss it with them on whether we should use their npm module directly or use something different like request-light.

By the way, I forgot to ask this before, why did we choose to use request-light instead of reusing jsforce? jsforce should be able to send http requests as well to the different endpoints.

CC: @chowwinston since he is also doing work with integration tests and calls to jsforce. This is not a simple jsforce vs request-light, but I want to understand if we can just use one to keep the code base more consistent.

Reply:
Okay, I'll note what we use the CLI for and their alternatives.

It looks like RequestStarted event has that url in its description field, so I can get that into the view without fetching ApexDebuggerRequest. I'll play around with it to see if it's too crowded in the callstack view. At the very least, it'll be logged in the debug console.

I picked request-light because I don't need all the other stuff that come with jsforce. If we want to just use jsforce, I think Winston would have to move the jsforce dependency to salesforcedx-utils-vscode so both salesforcedx-apex-debugger & salesforcedx-sobjects can use it.

Reply:
Here's what it looks like with the request's description in the callstack view:

![callstacknarrow](https://user-images.githubusercontent.com/7286437/29990710-1fd99b34-8f32-11e7-82c9-88bf327f0744.png)

![callstackwide](https://user-images.githubusercontent.com/7286437/29990711-234a01aa-8f32-11e7-8436-847d17bc85aa.png)

The callstack view is restricted to the left hand side, so the description is not as useful. I'm going to stick with just showing the request ID and outputting the request description in debug console for now. The latter is taken care of in https://github.com/forcedotcom/salesforcedx-vscode/pull/94.

Reply:
@chowwinston's jsforce dependency is currently in his new salesforcedx-sobject module. See #96. 

I think we can decide on whether to consolidate in a different PR.
For now, you can continue to use request-light first. 
Don't let this issue block you for now.